controllers/comments-controllers.js

```diff
@@ -24,6 +24,10 @@ exports.postComment = async (req, res, next) => {
     const { article_id: id } = req.params;
     const { username, body } = req.body;
 
+    // check if exists, if not throw 404
+    await fetchArticle(id);
+    // Promise.all doesn't work here, makeComment has a chance of throwing a 400 "violates foreign key constraint" first
```

Comment:
You may find that if the article id doesn't exist then you'll have a similar issue - which is actually a good thing! Just catch the error and pass it to `next`